lib/livebook/live_markdown/import.ex

```diff
@@ -394,15 +394,26 @@ defmodule Livebook.LiveMarkdown.Import do
 
   defp postprocess_notebook(notebook) do
     sections =
-      Enum.map(notebook.sections, fn section ->
+      Enum.with_index(notebook.sections)
+      |> Enum.map(fn {section, section_idx} ->
         # Set parent_id based on the persisted branch_parent_index if present
         case section.parent_id do
           nil ->
             section
 
           {:idx, parent_idx} ->
-            parent = Enum.at(notebook.sections, parent_idx)
-            %{section | parent_id: parent.id}
+            cond do
+              section_idx > parent_idx && Enum.at(notebook.sections, parent_idx) != nil ->
+                parent = Enum.at(notebook.sections, parent_idx)
```

Comment:
We should also verify that the parent doesn't have a parent section. Maybe something like `map_reduce` and keep track of section ids that have parent or something like that :)

Reply:
thanks for the input.

My solution though has to iterate 2 times the sections list, which i dislike a bit but i couldn't think a way to avoid it.
Let me know if you have a better solution

Reply:
I was thinking something like this:

```elixir
notebook.sections
|> Enum.with_index()
|> Enum.map_reduce({MapSet.new(), warnings}, fn {section, idx}, {branching_ids, warnings} ->
  case section.parent_id do
    nil -> ...

    {:id, parent_idx} ->
      parent = Enum.at(notebook.sections, parent_idx)

      cond do
        parent_idx >= idx ->
          "warn not before"

        MapSet.member?(branching_ids, parent.id) ->
          "warn parent already branching"

        true ->
          {%{section | parent_id: parent.id}, {MapSet.put(branching_ids, section.id), warnings}}
      end
  end
end)
```

Reply:
Ah, sorry I missed the comment. We definitely only want to skip the parent annotation and keep the section :)